src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```
@@ -460,45 +472,258 @@ public void inferenceSentenceHighlighting(
+            listener.onFailure(new IllegalArgumentException("Unsupported agent type: " + agentType));
+            return;
+        }
+
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("question", agenticQuery.getQueryText());
+
+        // Add index names if present
+        String[] indices = request.indices();
+        if (indices != null && indices.length > 0) {
+            Object indexValue = (type == MLAgentType.FLOW) ? indices[0] : indices;
+            parameters.put("index_name", indexValue.toString());
+        }
+
+        if (agenticQuery.getQueryFields() != null && !agenticQuery.getQueryFields().isEmpty()) {
+            parameters.put("query_fields", gson.toJson(agenticQuery.getQueryFields()));
+        }
+
+        if (hasSystemPrompt == false && type != MLAgentType.FLOW) {
+            parameters.put("system_prompt", loadSystemPrompt());
+        }
+
+        if (hasUserPrompt == false && type != MLAgentType.FLOW) {
+            parameters.put("user_prompt", loadUserPrompt());
+        }
+
         RemoteInferenceInputDataSet dataset = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
         AgentMLInput agentMLInput = new AgentMLInput(agentId, null, FunctionName.AGENT, dataset);
+
         mlClient.execute(FunctionName.AGENT, agentMLInput, ActionListener.wrap(response -> {
+            MLOutput mlOutput = (MLOutput) response.getOutput();
+            String result = null;
+            if (type == MLAgentType.FLOW) {
+                result = extractFlowAgentResult(mlOutput);
+            } else if (type == MLAgentType.CONVERSATIONAL) {
+                Map<String, String> conversationalResult = extractConversationalAgentResult(mlOutput, xContentRegistry);
+                result = conversationalResult.get("dsl_query");
+            } else {
+                listener.onFailure(new IllegalArgumentException("Unsupported agent type: " + agentType));
+                return;
```

Should we fail early for this case before executing the agent? Because other agents like Conversational Flow and PER might fail given these parameters.

---

Good call